red/reporter.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2015 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+var util = require('util');
+var EventEmitter = require('events').EventEmitter;
+var Log = require("./log");
+var TEN_SECONDS_MILLIS = 10000;
+
+var reporter = {
+    init: function(settings) {
+        if(settings) {
+            // TODO settings might be used in the future for configuring runtime reports
+        }
+        startInterval(TEN_SECONDS_MILLIS);
+    }
```

Comment:
It is needed because node-red has a life-cycle that must be honoured. It gets initialised, it gets started, it gets stopped. Nothing should be 'running' before it is started. Nothing should be 'running' after it is stopped.